### PR TITLE
Add `include_task_info` to action DB fetching operations

### PR DESCRIPTION
**Pull request by Dany9966**
-  Adds `include_task_info` option to all DB API action fetching methods that
    will control the loading of `info` column for a `BaseTransferAction`.
    Also adds `to_dict` option that turns the model objects into dictionaries
    for GET requests. This is necessary because serialization of model objects
    with deferred columns is not possible without an active DB session.

 -  Expose `include_task_info` parameters to `include_tasks_info_in_api` config options

### CODE REVIEW

File: coriolis/conductor/rpc/server.py [hunk 1]
```diff
@@ -1029,14 +1029,18 @@ def execute_replica_tasks(self, ctxt, replica_id, shutdown_instances):
 
     @replica_synchronized
     def get_replica_tasks_executions(self, ctxt, replica_id,
-                                     include_tasks=False):
+                                     include_tasks=False, include_info=False,
+                                     to_dict=True):
```
Comment (aznashwan) [inline coriolis/conductor/rpc/server.py:1033]: Is there any reason for adding the `to_dict` kwarg here and not using it anywhere else?

If it's a pickling issue and we can't send a DB object over oslo_messaging then we should always set this to `True` at the DB layer IMO and never propagate the non-dict objects down the stack.

File: coriolis/conductor/rpc/server.py [hunk 2]
```diff
@@ -1029,14 +1029,18 @@ def execute_replica_tasks(self, ctxt, replica_id, shutdown_instances):
...
         return db_api.get_replica_tasks_executions(
-            ctxt, replica_id, include_tasks)
+            ctxt, replica_id, include_tasks, include_info=include_info,
+            to_dict=to_dict)
 
     @tasks_execution_synchronized
-    def get_replica_tasks_execution(self, ctxt, replica_id, execution_id):
+    def get_replica_tasks_execution(self, ctxt, replica_id, execution_id,
+                                    include_info=False, to_dict=True):
```
Comment (aznashwan) [inline coriolis/conductor/rpc/server.py:1040]: Ditto `to_dict`.

File: coriolis/conductor/rpc/server.py [hunk 3]
```diff
@@ -1066,21 +1070,27 @@ def cancel_replica_tasks_execution(self, ctxt, replica_id, execution_id,
...
         if not execution:
             raise exception.NotFound(
                 "Execution with ID '%s' for Replica '%s' not found." % (
                     execution_id, replica_id))
         return execution
 
-    def get_replicas(self, ctxt, include_tasks_executions=False):
-        return db_api.get_replicas(ctxt, include_tasks_executions)
+    def get_replicas(self, ctxt, include_tasks_executions=False,
+                     include_info=False, to_dict=True):
```
Comment (aznashwan) [inline coriolis/conductor/rpc/server.py:1085]: Ditto `to_dict`.

File: coriolis/conductor/rpc/server.py [hunk 4]
```diff
@@ -1066,21 +1070,27 @@ def cancel_replica_tasks_execution(self, ctxt, replica_id, execution_id,
...
+        return db_api.get_replicas(
+            ctxt, include_tasks_executions, include_info=include_info,
+            to_dict=to_dict)
 
     @replica_synchronized
-    def get_replica(self, ctxt, replica_id):
-        return self._get_replica(ctxt, replica_id)
+    def get_replica(self, ctxt, replica_id, include_info=False, to_dict=True):
+        return self._get_replica(
+            ctxt, replica_id, include_info=include_info, to_dict=to_dict)
```
Comment (aznashwan) [inline coriolis/conductor/rpc/server.py:1093]: `to_dict`

File: coriolis/conductor/rpc/server.py [hunk 5]
```diff
@@ -1196,23 +1206,26 @@ def create_instances_replica(self, ctxt, origin_endpoint_id,
         LOG.info("Replica created: %s", replica.id)
         return self.get_replica(ctxt, replica.id)
 
-    def _get_replica(self, ctxt, replica_id):
-        replica = db_api.get_replica(ctxt, replica_id)
+    def _get_replica(self, ctxt, replica_id, include_info=False, to_dict=False):
+        replica = db_api.get_replica(
+            ctxt, replica_id, include_info=include_info, to_dict=to_dict)
```
Comment (aznashwan) [inline coriolis/conductor/rpc/server.py:1211]: `to_dict`

File: coriolis/conductor/rpc/server.py [hunk 6]
```diff
@@ -1196,23 +1206,26 @@ def create_instances_replica(self, ctxt, origin_endpoint_id,
...
-            include_info=include_info)
+            include_info=include_info,
+            to_dict=to_dict)
 
     @migration_synchronized
-    def get_migration(self, ctxt, migration_id):
-        # the default serialization mechanism enforces a max_depth of 3
-        return utils.to_dict(self._get_migration(ctxt, migration_id))
+    def get_migration(self, ctxt, migration_id, include_info=False,
+                      to_dict=True):
```
Comment (aznashwan) [inline coriolis/conductor/rpc/server.py:1226]: `to_dict`

File: coriolis/conductor/rpc/server.py [hunk 7]
```diff
@@ -2086,8 +2099,10 @@ def migrate_instances(self, ctxt, origin_endpoint_id,
 
         return self.get_migration(ctxt, migration.id)
 
-    def _get_migration(self, ctxt, migration_id):
-        migration = db_api.get_migration(ctxt, migration_id)
+    def _get_migration(self, ctxt, migration_id, include_info=False,
+                       to_dict=False):
+        migration = db_api.get_migration(
+            ctxt, migration_id, include_info=include_info, to_dict=to_dict)
```
Comment (aznashwan) [inline coriolis/conductor/rpc/server.py:2105]: `to_dict`

**Review by aznashwan — COMMENTED** [review]
Mostly LGTM. I'd like to request you change `include_info` to something a bit more suggestive like `include_transfer_info`.

File: coriolis/db/api.py [hunk 8]
```diff
@@ -497,22 +515,22 @@ def get_replica_migrations(context, replica_id):
 
 @enginefacade.reader
 def get_migrations(context, include_tasks=False,
-                   include_info=False, to_dict=True):
```
Comment (aznashwan) [inline coriolis/db/api.py:500]: Can you please make triple-sure that the `to_dict` kwargs' default changing doesn't cause issues with any other API call (most notably, the ones performed by the UI?

**Comment by aznashwan** [comment]
Second review pass done, all ll LGTM now, should be test-able.

File: coriolis/db/api.py (hunk 8 above)
Comment (Dany9966) [reply coriolis/db/api.py:500]: This method is only used when listing migrations, and I have `to_dict` explicitly set to True there. No UI issues.

**Comment by gluka-CBS** [comment]
tested and works great. Great improvement on the UI with PMR/large disk combo

**Comment by aznashwan** [comment]
Merging, thanks!

**Pull request merged**